### DISCUSSION TITLE
getting message about failed acknowledgement when such is not enabled

### DISCUSSION THREAD

**Discussion by tshepang** [vectordotdev/vector#13433] [Q&A] [answered]
I have this log:

> 2022-07-05T05:44:16.908503Z ERROR source{component_kind="source" component_id=logstash-in component_type=logstash component_name=logstash-in}:connection{peer_addr=<ip addr>:35754}: vector::internal_events::tcp: Error writing acknowledgement, dropping connection. error=Broken pipe (os error 32) error_code="ack_failed" error_type="writer_failed" stage="sending" internal_log_rate_secs=10

Looking at docs, one has to explicitly enable acknowledgements, so makes me wonder if this is a false message.

**Comment by jszwedko** [comment, marked as answer]
Hi @tshepang !

I can understand the confusion here. Even when acknowledgements are not enabled in Vector, the `logstash` source still has to write acknowledgements to the client to satisfy the Logstash protocol. The difference is, when `acknowledgements` are enabled for a sink connected to a `logstash` source, Vector will wait to write the acknowledgement until the sink has processed the associated events. When `acknowledgements` are not enabled for a connected sink, the `logstash` source just immediately acknowledges events once it decodes them.

**Reply by spencergilbert** [reply]
Feels like we should edit the error message to clarify it's the logstash protocol here vs Vector's acking.
